setup.py

```diff
@@ -18,131 +18,67 @@
 #  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 
+# NOTE: The configuration for the package, including the name, version, and
+# other information are set in the setup.cfg file.
+
+import os
 import sys
 
-# python_requires will stop pip, but also let users who are using
-# 'python setup.py develop'.
-#
-if sys.version_info < (3, 5):
-    sys.stderr.write("Sherpa 4.12 (and later) requires Python 3.5 or later.\n\n")
-    sys.stderr.write("Please use Sherpa 4.11.1 if you need to use Python 2.7\n")
-    sys.exit(1)
+from setuptools import setup
 
-try:
-    import setuptools
-except:
-    print((
-        "WARNING\n"
-        "Could not import setuptools.\n"
-        "This might lead to an incomplete installation\n"
-    ), file=sys.stderr)
+from helpers.extensions import static_ext_modules
 
-# The module used to try to find the numpy module and error out if
-# that could not be found, but it seems simpler to just error out
-# here.
-#
-try:
-    from numpy.distutils import core
+# First provide helpful messages if contributors try and run legacy commands
+# for tests or docs.
 
-except ImportError:
-    import sys
+TEST_HELP = """
+Note: running tests is no longer done using 'python setup.py test'. Instead
+you will need to run:
+    tox -e test
+If you don't already have tox installed, you can install it with:
+    pip install tox
+If you only want to run part of the test suite, you can also use pytest
+directly with::
+    pip install -e .[test]
+    pytest
+For more information, see:
+  http://docs.astropy.org/en/latest/development/testguide.html#running-tests
+"""
 
-    print((
-        "You need to install NUMPY in order to build Sherpa\n"
-        "Other dependencies will be automatically installed\n"
-        "Please install NUMPY (e.g. pip install numpy) and try again."
-    ), file=sys.stderr)
-    sys.exit(2)
+if 'test' in sys.argv:
+    print(TEST_HELP)
+    sys.exit(1)
 
-from helpers.extensions import static_ext_modules
+DOCS_HELP = """
```

Comment:
A minor issue, but the existing documentation only talks about using `build_sphinx` and not `build_docs` - i.e. https://sherpa.readthedocs.io/en/latest/install.html#building-the-documentation

Do we want this message to note the `build_sphinx` option as well?